src/ert/storage/local_experiment.py

```diff
@@ -335,6 +335,43 @@ def response_key_to_response_type(self) -> Dict[str, str]:
         mapping = {}
         for config in self.response_configuration.values():
             for key in config.keys:
+                if key == "*":
+                    continue
+
                 mapping[key] = config.response_type
 
         return mapping
+
+    def update_response_keys(
+        self, response_type: str, new_response_keys: List[str]
+    ) -> None:
+        if not any(
+            k for k in new_response_keys if k not in self.response_key_to_response_type
+        ):
+            return None
+
+        responses_configuration = self.response_configuration
+        if response_type not in responses_configuration:
+            raise KeyError(
+                f"Response type {response_type} does not exist in current responses.json"
+            )
+
+        config = responses_configuration[response_type]
+
+        diff = set(new_response_keys) - set(config.keys)
+
+        if len(diff) > 0:
```

Comment:
Nitpick, but could also write:
```suggestion
        if diff:
```

and perhaps rename the variable for clarity? Think I would have used `new_response_keys` for this, and the variable you already have named that would be more like: `response_keys`?

Then it becomes:
```python
if new_response_keys:
```
which to me is more readable.